### PR TITLE
[VL] Improving performance by using simd gather instructions 

### PR DESCRIPTION
**Pull request by zuochunwei**
## What changes were proposed in this pull request?

please don't review this PR now, wait for me to make a test.

(Please fill in changes proposed in this fix)

(Fixes: \#ISSUE-ID)

## How was this patch tested?

(Please explain how this patch was tested. E.g. unit tests, integration tests, manual tests)

(If this patch involves UI changes, please attach a screenshot; otherwise, remove this)

### CODE REVIEW

**Comment by zuochunwei** [comment]
The result does not conform to the expectation, and I will close later.

**Comment by FelixYBW** [comment]
No benefit but no hurt either. We may keep it if you like

File: cpp/CMakeLists.txt [hunk 1]
```diff
@@ -83,6 +83,7 @@ add_compile_options(-Werror)
 add_compile_options(-Wno-error=parentheses)
 add_compile_options(-Wno-error=unused-function)
 add_compile_options(-Wno-error=unused-variable)
+add_compile_options(-mavx2)
```
Comment (ulysses-you) [inline cpp/CMakeLists.txt:86]: macos arm64 does not support this option, can we only enable it for linux os ?

**Pull request closed without merging**